src/modules/sharing/sharing.service.ts

```diff
@@ -1799,4 +1787,67 @@ export class SharingService {
         });
     }
   }
+
+  async changeSharingType(
+    user: User,
+    itemId: Sharing['itemId'],
+    itemType: Sharing['itemType'],
+    type: Sharing['type'],
+  ): Promise<void> {
+    let item: File | Folder;
+
+    if (itemType === 'file') {
+      item = await this.fileUsecases.getByUuid(itemId);
+    } else if (itemType === 'folder') {
+      item = await this.folderUsecases.getByUuid(itemId);
+    }
+    const isUserOwner = item.isOwnedBy(user);
+
+    if (!isUserOwner) {
+      throw new ForbiddenException();
+    }
+
+    const oldSharingType =
+      type === SharingType.Private ? SharingType.Public : SharingType.Private;
+
+    await this.sharingRepository.updateSharing(
+      {
+        itemId,
+        itemType,
+        sharedWith: '00000000-0000-0000-0000-000000000000',
```

Comment:
If this is not public, how are we going to use this as a condition in any case? Will not break when the the sharing is private?

Reply:
If is not public, it just doesn't try to get roles or something like that for sharings with 0000-0000. 

It does not break when the sharing is private. At least not with our current implementation, which is mainly using 
`findSharingsWithRolesByItem` to fetch private users and their roles.

- We could add some flag to deactivate the link without changing it to private if we dont want the link to be invalidated between type changes. 
- Deleting it is an option too, but given that case, the previous LINK is going to be invalidated.